consensus/src/chained_bft/safety/safety_rules.rs

```diff
@@ -245,6 +246,8 @@ impl<T: Payload> SafetyRules<T> {
                 // 1) B0 <- B1 <- B2 <--
                 // 2) round(B0) + 1 = round(B1), and
                 // 3) round(B1) + 1 = round(B2).
+                checked_precondition!(three_chain_head.round() < std::u64::MAX);
```

Comment:
Hiding a precondition inside conditional like this is not good style. It also seems to me that given the precondition that one_chain_head.round() < std::u64::MAX, it should be possible to derive from that and invariants that three_chain_head.round() < std::u64::MAX, so this need to be called out as precondition in its own right.